core/src/main/java/org/apache/calcite/rex/RexLiteral.java

```diff
@@ -551,12 +552,16 @@ private static int width(TimeUnit timeUnit) {
    * Prints the value this literal as a Java string constant.
    */
   public void printAsJava(PrintWriter pw) {
-    printAsJava(value, pw, typeName, true, RexDigestIncludeType.NO_TYPE);
+    try {
+      appendAsJava(value, pw, typeName, true, RexDigestIncludeType.NO_TYPE);
+    } catch (IOException e) {
+      throw new IllegalStateException("IOException should not be raised with PrintWriter", e);
```

Comment:
UncheckedIOException?

Reply:
Passing a PrintWriter should not raise an IOException, that's why I preferred IllegalState in this part.

Reply:
I like that way of thinking. We could state in the method javadoc that we don't allow an `Appendable` that might actually do I/O. (The goal is only to write to in-memory buffers.) And therefore if there is an `IOException` it is a coding error.